### PR TITLE
web-wallet: Add Existing Wallet Notice

### LINKED ISSUE #1360: web-wallet: Add notice on login if local storage is about to get reset (ITN)
#### Summary
When logging in to a wallet, the local data gets cleared if the new wallet does not match the most recent one the user has used. This leads to longer waiting times for resync to happen. Furthermore, users might not be aware that running two wallets at the same time on the same browser is not supported.

We should introduce a notice during log-in, in case local data exists and will get deleted, due to the user using another wallet to log in.

This should be included in the restore flow, as well as on the main login page.

**Design:**
[Figma](https://www.figma.com/file/ImsgL6Xf0R31HFUIw2xxnw/Wallet-Extension-(FEB-08)?type=design&node-id=3264%3A80830&mode=design&t=iKsNUnm1GF5ZKUhz-1)

Screenshot 2024-02-09 at 14 00 51

### PR DESCRIPTION
**Pull request by nortonandreev**
Resolves #1360

### CODE REVIEW

File: web-wallet/src/routes/(welcome)/login/__tests__/page.spec.js [hunk 1]
```diff
@@ -120,6 +191,8 @@ describe("Login", async () => {
 
 			await fireEvent.input(textInput, { target: { value: mnemonic } });
 			await fireEvent.submit(form, { currentTarget: form });
+
+			expect(confirmSpy).not.toBeCalled();
```
Comment (ascartabelli) [inline web-wallet/src/routes/(welcome)/login/__tests__/page.spec.js:195]: See above about matchers.

File: web-wallet/src/routes/(welcome)/login/__tests__/page.spec.js [hunk 2]
```diff
@@ -172,6 +245,8 @@ describe("Login", async () => {
 
 			await fireEvent.input(textInput, { target: { value: `  \t${mnemonic} \t  ` } });
 			await fireEvent.submit(form, { currentTarget: form });
+
+			expect(confirmSpy).not.toBeCalled();
```
Comment (ascartabelli) [inline web-wallet/src/routes/(welcome)/login/__tests__/page.spec.js:249]: See above about matchers.

File: web-wallet/src/routes/(welcome)/setup/create/+page.svelte [hunk 3]
```diff
@@ -41,12 +47,19 @@
 	/** @type {string[]} */
 	let enteredMnemonicPhrase = [];
 
+	const collectSettings = getKey("userId");
+	const userId = collectSettings($settingsStore);
```
Comment (ascartabelli) [inline web-wallet/src/routes/(welcome)/setup/create/+page.svelte:51]: As you're retrieving properties as they are, you may want to just use destructuring here:

```js
const { userId } = $settingsStore;
```

File: web-wallet/src/routes/(welcome)/setup/restore/+page.svelte [hunk 4]
```diff
@@ -28,12 +34,19 @@
 	/** @type {string[]} */
 	let mnemonicPhrase = [];
 
+	const collectSettings = getKey("userId");
```
Comment (ascartabelli) [inline web-wallet/src/routes/(welcome)/setup/restore/+page.svelte:37]: See above about destructuring.

File: web-wallet/src/routes/(welcome)/setup/restore/+page.svelte [hunk 5]
```diff
@@ -28,12 +34,19 @@
...
+	const userId = collectSettings($settingsStore);
+
 	$: if (showPasswordSetup) {
 		password = showPasswordSetup ? password : "";
 	}
 </script>
 
-{#if !tosAccepted}
+{#if !notice && userId.length !== 0}
```
Comment (ascartabelli) [inline web-wallet/src/routes/(welcome)/setup/restore/+page.svelte:45]: As we said yesterday `userId` might not exist, in theory, if the user comes from an older version of the web-wallet.
We can just check that `userId` is not a "falsy" value here (as you did in the login page with `currentAddress`.

**Review by ascartabelli — CHANGES_REQUESTED** [review]
Left just some very very minor notes.
I'd like to have some additional tests for the create and restore flows, though: as we're in a rush today and everyone is sick, please open an issue for that so we can tackle it later on.

**Review by ascartabelli — APPROVED** [review]

**Review by Fulviuus — APPROVED** [review]

**Pull request merged**